modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -195,10 +196,10 @@ public void onWrite(Iterator<CommandClosure<WriteCommand>> iterator) {
 
                 assert safeTimePropagatingCommand.safeTime() != null;
 
-                safeTime.update(safeTimePropagatingCommand.safeTime().asHybridTimestamp());
+                updateTrackerWithIgnoreTrackerClosedException(safeTime, safeTimePropagatingCommand.safeTime().asHybridTimestamp());
             }
 
-            storageIndexTracker.update(commandIndex);
+            updateTrackerWithIgnoreTrackerClosedException(storageIndexTracker, commandIndex);
```

Comment:
`Replica#waitForActualState()` waits for next index. It might happen that the `PartitionListener` actually executes the command and updates the index in the storage, but then fails to update the tracker (and notify the subscribers, including `Replica#waitForActualState()`). Would it be handled correctly in `Replica#waitForActualState()`? Maybe we need to call someone from the TX team.

Reply:
The question is good, as a result, errors are processed here `org.apache.ignite.internal.replicator.ReplicaManager#onPlacementDriverMessageReceived`, but I still don’t know how it should be.

Reply:
Discussed everything together in person, decided that so far this is not a problem.